### PR TITLE
build(deps): mobx@6 upgrade

### PR DESCRIPTION
**Pull request by erickzhao**
Bumps up our `mobx` and `mobx-react` dependencies to 6.x.y. This is part of a wider attempt to modernize our renderer codebase, since we're stuck on old versions of React and using class components.

### Codemod

In order to perform this upgrade, I followed MobX's [Migrating from MobX 4/5](https://mobx.js.org/migrating-from-4-or-5.html) guide.

Because we use `@observer` decorators extensively, I decided to use the [`npx mobx-undecorate` codemod](https://www.npmjs.com/package/mobx-undecorate) provided by the guide. This will allow us to immediately upgrade to MobX 6 while keeping our React component architecture.

We can then perform a piecewise upgrade of our React components in the future.

### Caveats

* The types for the new `observer()` higher-order component from MobX mean that we use `any` types when mounting Enzyme components. This is a trade-off I'm willing to make because Enzyme doesn't support React 17 yet and we might want to look towards migrating towards another unit testing library.
* This diff is huge. Reviewing it will be easier with the "hide whitespace" option.

### CODE REVIEW

File: tests/renderer/components/commands-address-bar-spec.tsx [hunk 1]
```diff
@@ -31,7 +31,7 @@ describe('AddressBar component', () => {
 
   it('handles change', () => {
     const wrapper = shallow(<AddressBar appState={store as any} />);
-    const instance: AddressBar = wrapper.instance() as any;
+    const instance: any = wrapper.instance() as any;
```
Comment (BlackHole1) [inline tests/renderer/components/commands-address-bar-spec.tsx:34]:
```suggestion
    const instance = wrapper.instance() as any;
```

File: tests/renderer/components/commands-address-bar-spec.tsx [hunk 2]
```diff
@@ -48,7 +48,7 @@ describe('AddressBar component', () => {
   it('handles submit', () => {
     const preventDefault = jest.fn();
     const wrapper = shallow(<AddressBar appState={store as any} />);
-    const instance: AddressBar = wrapper.instance() as any;
+    const instance: any = wrapper.instance() as any;
```
Comment (BlackHole1) [inline tests/renderer/components/commands-address-bar-spec.tsx:51]: Above

**Review by BlackHole1 — CHANGES_REQUESTED** [review]

**Comment by BlackHole1** [comment]
Perhaps it would be better if we used the following?

``` ts
// AddThemeDialog.tsx
export class AddThemeDialogComponent {
  // code
}
export const AddThemeDialog = observer(AddThemeDialogComponent);


// AddThemeDialog-spec.tsx
const wrapper = shallow(<AddThemeDialog appState={store as any} />);
const instance = wrapper.instance() as AddThemeDialogComponent;
```

**Review by ckerr — COMMENTED** [review]
Even in "hide whitespace" mode, it seems like a lot of this diff is caused by line wrap changes :roll_eyes: This PR isn't as scary as it initially looked

**Review by ckerr — APPROVED** [review]
Looks fine to me. The only thing I don't like abou this PR is the reversion from specific types to `any` but as per the description that seems to be unavoidable for now.

I don't have any extra change requests that weren't already discussed by @BlackHole1

**Comment by erickzhao** [comment]
FWIW the `:any` changes in our specs will likely be solved with the migration from Enzyme to RTL: https://github.com/electron/fiddle/pull/1064

**Pull request merged**